packages/opal-client/opal_client/data/fetcher.py

```diff
@@ -51,8 +52,11 @@ async def stop(self):
         """Release internal tasks and resources."""
         await self._engine.terminate_workers()
 
-    async def handle_url(self, url, config):
+    async def handle_url(self, url, config, data):
```

Comment:
if both data and url is none this is not valid and we should ignore and issue warning

Reply:
data wins